src/toast/Toast.tsx

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import {Icon} from "../icon";
+import {classNames} from "../utils";
+import {ClassNames} from "../modals/ClassNames";
+
+export enum Direction {
+    TOP_RIGHT = "top-right",
+    BOTTOM_LEFT = "bottom-left",
+    BOTTOM_RIGHT = "bottom-right",
+    TOP_LEFT = "top-left",
+}
+
+export enum MessageType {
+    SUCCESS = "success",
+    WARNING = "warning",
+    DANGER = "danger",
+    INFO = "info",
+}
+
+const DEFAULT_HIDE_TIME = 3000;
+
+type ToastProps = {
+    showToast?: boolean;
+    duration?: number;
+    position?: Direction;
+    type?: MessageType;
+    icon?: string;
+    text: string;
+    onClose?: Function;
+    style?: any;
+    className?: string;
+};
+
+type ToastState = {
+    showToast: boolean;
+};
+
+export class Toast extends React.PureComponent<ToastProps> {
+    private divRef: HTMLDivElement | null = null;
+    // Remove the warning for setState on componentUnmount using isMounted
+    private _isMounted = false;
+
+    state: ToastState = {
+        showToast: this.props.showToast !== undefined ? this.props.showToast : false,
+    };
+
+    // Default value for props if no value is given
+    static defaultProps = {
+        showToast: false,
+        type: MessageType.SUCCESS,
+        position: Direction.TOP_RIGHT,
+        duration: DEFAULT_HIDE_TIME,
+    };
+
+    componentWillUpdate(nextProps: ToastProps, nextState: ToastState) {
+        if (this.showToast(nextProps, nextState)) {
+            this.createElement();
+
+            if (nextProps.showToast === true && nextProps.showToast !== nextState.showToast) {
+                //set the timer for the toast
+                this.setTimer(nextProps.duration!);
+            }
+        } else {
+            this.cleanup();
+        }
+    }
+
+    componentWillMount() {
```

Comment:
Same comment from Modal.tsx.  Why do we need this?  Can't we just rely on componentWillUpdate to create the placeholder <div/>?

Reply:
I was initially facing similar issue in Toast as well.

Reply:
Okay, got it.  Thanks.